src/parameters/division.rs

```diff
@@ -0,0 +1,53 @@
+use super::PywrError;
+use crate::metric::Metric;
+use crate::model::Model;
+use crate::parameters::{Parameter, ParameterMeta};
+use crate::scenario::ScenarioIndex;
+use crate::state::State;
+use crate::timestep::Timestep;
+use std::any::Any;
+use crate::PywrError::InvalidMetricValue;
+
+
+pub struct DivisionParameter {
+    meta: ParameterMeta,
+    numerator: Metric,
+    denominator: Metric
+}
+
+impl DivisionParameter {
+    pub fn new(name: &str,  numerator: Metric, denominator: Metric) -> Self {
+        Self {
+            meta: ParameterMeta::new(name),
+            numerator,
+            denominator
+        }
+    }
+}
+
+impl Parameter for DivisionParameter {
+    fn as_any_mut(&mut self) -> &mut dyn Any {
+        self
+    }
+    fn meta(&self) -> &ParameterMeta {
+        &self.meta
+    }
+    fn compute(
+        &self,
+        _timestep: &Timestep,
+        _scenario_index: &ScenarioIndex,
+        model: &Model,
+        state: &State,
+        _internal_state: &mut Option<Box<dyn Any + Send>>,
+    ) -> Result<f64, PywrError> {
+        // TODO handle scenarios
```

Comment:
I've just had a quick look at this.

I think there's some syntax errors here to do with the float constants. 

Pywr's current behaviour will return and unchecked division which will be a NaN. I think we should probably catch that here, like your implementation. There is a check for NaN in the parent code that calls `.compute()`, but it is a bit crude at the moment. 

Something like this:
```rust
        let denominator = self.denominator.get_value(model, state)?;

        if denominator == 0.0 {
            return Err(InvalidMetricValue(format!("Division by zero creates a NaN."));
        }

        let numerator = self.numerator.get_value(model, state)?;
        Ok(numerator / denominator)
```